docs/network.md

```diff
@@ -0,0 +1,217 @@
+They are kept in `edge_info` data structure of type `HashSet<Edge>`.
+
+```rust
+pub struct RoutingTableActor {
+    /// Collection of edges representing P2P network.
+    /// It's indexed by `Edge::key()` key and can be search through by called `get()` function
+    /// with `(PeerId, PeerId)` as argument.
+    pub edges_info: HashSet<Edge>,
+    /// ...
+}
+```
+
+# 6. Code flow - connecting to a peer - handshake
+
+When `PeerManagerActor` starts it starts to listen to a specific port.
+
+## 6.1 - Step 1 - `monitor_peers_trigger` runs
+`PeerManager` checks if we need to connect to another peer by running `PeerManager::is_outbound_bootstrap_needed` method.
+If `true` we will try to connect to new node.
+Let's call current node, node `A`.
+
+## 6.2 - Step 2 - choosing node to connect to
+Method `PeerManager::sample_random_peer` will be called, and it returns node `B` that we will try to connect to.
+
+## 6.3 - Step 3 - `OutboundTcpConnect` message
+`PeerManagerActor` will send to itself a message `OutboundTcpConnect` in order to connect to node `B`.
+
+```rust
+pub struct OutboundTcpConnect {
+    /// Peer information of the outbound connection
+    pub target_peer_info: PeerInfo,
+}
+```
+
+## 6.4 - Step 4 - `OutboundTcpConnect` message
+On receiving the message `handle_msg_outbound_tcp_connect` method will be called, which calls
+`TcpStream::connect` to create new connection.
+
+## 6.5 - Step 5 - Connection gets established
+Once connection with outgoing peer gets established.
+`try_connect_peer` method will be called.
+And then new `PeerActor` will be created and started.
+One `PeerActor` starts it will send `Handshake` message to outgoing node `B` over tcp connection.
+
+This message contains `protocol_version`, node's `A` metadata, as well as all information necessary to create `Edge`.
+
+```rust
+pub struct Handshake {
+    /// Current protocol version.
+    pub(crate) protocol_version: u32,
+    /// Oldest supported protocol version.
+    pub(crate) oldest_supported_version: u32,
+    /// Sender's peer id.
+    pub(crate) sender_peer_id: PeerId,
+    /// Receiver's peer id.
+    pub(crate) target_peer_id: PeerId,
+    /// Sender's listening addr.
+    pub(crate) sender_listen_port: Option<u16>,
+    /// Peer's chain information.
+    pub(crate) sender_chain_info: PeerChainInfoV2,
+    /// Represents new `edge`. Contains only `none` and `Signature` from the sender.
+    pub(crate) partial_edge_info: PartialEdgeInfo,
+}
+```
+
+## 6.6 - Step 6 - `Handshake` arrives at node `B`
+
+Node `B` receives `Handshake` message. Then performs various validation checks.
+
+If everything is successful, `PeerActor` will send `Consolidate` message to `PeerManagerActor`.
+This message contains everything needed to add `PeerActor` to list of active connections in `PeerManagerActor`.
+
+Otherwise, `PeerActor` will be stopped immediately or after some timeout.
+
+TODO: Rename `Consolidate` to `RegisterPeer` + document all attributes.
```

Comment:
Could we link an issue here

Reply:
We already merged that PR. It's already renamed.